relay-server/src/actors/project.rs

```diff
@@ -823,12 +839,44 @@ mod tests {
 
     use relay_common::{ProjectId, ProjectKey, UnixTimestamp};
     use relay_metrics::{Bucket, BucketValue, Metric, MetricValue};
+    use relay_system::Service;
     use serde_json::json;
 
+    use crate::actors::envelopes::EnvelopeManager;
+    #[cfg(feature = "processing")]
+    use crate::actors::processor::EnvelopeProcessor;
+    use crate::actors::project_cache::ProjectCache;
+
     use super::{Config, Project, ProjectState, StateChannel};
 
-    #[test]
-    fn get_state_expired() {
+    macro_rules! empty_service {
+        ($service_name:ident, $interface:ident) => {
+            struct $service_name;
+
+            impl Service for $service_name {
+                type Interface = $interface;
+
+                fn spawn_handler(self, mut rx: relay_system::Receiver<Self::Interface>) {
+                    tokio::spawn(async move {
+                        loop {
+                            tokio::select! {
+                                Some(_) = rx.recv() => (),
+                                else => break,
+                            }
+                        }
+                    });
+                }
+            }
+        };
+    }
```

Comment:
Nice! We can probably put this in `testutils.rs`. Should we call this `dummy_service`?

Reply:
Instead of `testutils.rs`, put it in `relay_test/`. That way, we can also use it in other crates.

Also, I think you can write this easily without a macro. See the [original Gist](https://gist.github.com/jan-auer/68305125c13aaab7355a1c99b3b65012#file-main-rs-L214-L232).

Reply:
This also is not needed right now. 
As the followup I will look into this and add tests and also more test utils including suggestions.